airflow/www/security.py

```diff
@@ -441,14 +441,15 @@ def _merge_perm(self, permission_name, view_menu_name):
 
     def add_homepage_access_to_custom_roles(self):
         """
-        Add Website.can_read access to all roles.
+        Add Website.can_read access to all custom roles.
 
         :return: None.
         """
         website_permission = self.add_permission_view_menu(
             permissions.ACTION_CAN_READ, permissions.RESOURCE_WEBSITE
         )
-        for role in self.get_all_roles():
+        custom_roles = [role for role in self.get_all_roles() if role.name not in EXISTING_ROLES]
+        for role in custom_roles:
```

Comment:
Can you add a test to prevent regression -- that checks that only custom roles have `Website.can_read`

Reply:
@kaxil All roles except for Public will get `Website.can_read`. This is just only added to custom roles explicitly, since the default ones already have it. So mostly I just wanted to not give Public that access.

Still think it's worth adding a test to make sure that isn't available to Public?

Reply:
I think we should have a test somewhere that verifies the exact permissions the public role is supposed to have

Reply:
If we already have it, then not needed

Reply:
@jhtimmins Can you make sure to add test (if it does not exist) in a follow up PR or the other PR that you have open